### PR TITLE
coprocessor: migrate password function from TiDB

### LINKED ISSUE #6776: UCP: Migrate scalar function `Password` from TiDB
## Description

Port the scalar function `Password` from TiDB to coprocessor.

## Score 

* 50

## Mentor(s)

* @andylokandy 

## Recommended Skills

* Rust programming 

## Learning Materials

Already implemented expressions ported from TiDB

- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/rpn_expr) 
- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/expr)

### PR DESCRIPTION
**Pull request by psinghal20**
Signed-off-by: psinghal20 <psinghal20@gmail.com>

### What problem does this PR solve?

Issue Number: closes #6776

Problem Summary:

### What is changed and how it works?
Migrates the scalar function password from TiDB.

I am not sure of the correct way to report deprecation warning, let me know if any changes are required. @andylokandy PTAL

### Check List 

Tests 

- Unit test
- Integration test

### CODE REVIEW

**Comment by TennyZhuang** [comment]
LGTM

**Comment by TennyZhuang** [comment]
@Fullstop000  PTAL

File: components/tidb_query_normal_expr/src/builtin_encryption.rs [hunk 1]
```diff
@@ -384,4 +403,25 @@ mod tests {
...
+        ];
+        let mut ctx = EvalContext::default();
+
+        for (input_str, exp_str) in cases {
+            let input = datum_expr(Datum::Bytes(input_str.as_bytes().to_vec()));
+            let op = scalar_func_expr(ScalarFuncSig::Password, &[input]);
+            let op = Expression::build(&mut ctx, op).unwrap();
+            let got = op.eval(&mut ctx, &[]).unwrap();
+            let exp = Datum::Bytes(exp_str.as_bytes().to_vec());
+            assert_eq!(got, exp, "password('{:?}')", input_str);
```
Comment (Fullstop000) [inline components/tidb_query_normal_expr/src/builtin_encryption.rs:424]: It could be better to add testing for empty input and the warning I think :)

Comment (psinghal20) [reply components/tidb_query_normal_expr/src/builtin_encryption.rs:424]: @Fullstop000 I have added a Null case test, similar to the above tests. PTAL.

**Comment by Fullstop000** [comment]
@sre-bot /approve
